src/runtime/tuple_parser.c

```diff
@@ -224,30 +263,30 @@ static parser is_end_of_vector(struct _closure_is_end_of_vector *__self, parser
         completion vc = closure(h, value_complete, bound(t), intern_u64(*bound(index)), self);
         (*bound(index))++;
         // doesnt handle whitespace before end
-        return apply(ignore_whitespace(h, (void *)closure(h, parse_value, h, vc, bound(e))), in);
+        return apply(ignore_whitespace(h, (parser)closure(h, parse_value, h, vc, bound(e))), in);
     }
-    return apply(bound(c), bound(t));
+    parser p = apply(bound(c), bound(t));
+    deallocate_closure(self);
+    closure_finish();
+    return p;
 }
 
 static parser parse_value_string(struct _closure_parse_value_string *__self, character in)
 {
     heap h = bound(h);
-
     if (in == '"')
         return combinate(h, closure(h, quoted_string, h, bound(c), bound(b)));
 
-    parser p = combinate(h, closure(h, terminal, bound(c), value_terminal, bound(b)));
-    return apply(p, in);
+    return apply(combinate(h, closure(h, terminal, bound(c), value_terminal, bound(b))), in);
 }
 
-
+// leaks; no end condition to free on
 closure_function(0, 2, parser, kill,
```

Comment:
yeah, this has to live for the parser lifetime...so It could go there...I think this is just a drain on a syntax error?

Reply:
Perhaps building out an encompassing parser type would be order...though I think maybe for a future pass.